pkg/traces/servicegraphprocessor/processor.go

```diff
@@ -197,6 +247,9 @@ func (p *processor) Capabilities() consumer.Capabilities {
 func (p *processor) ConsumeTraces(ctx context.Context, td pdata.Traces) error {
 	level.Debug(p.logger).Log("msg", "consuming traces")
 
+	// Evict expired edges
+	p.expire()
```

Comment:
I thought of this too, but I'm not very convinced.

Running the function when consuming traces makes it scale with the throughput of the pipeline (i.e. more `ConsumeTraces` calls, more times it tries to expire). Finding the right amount of _expiry_ otherwise could be difficult.

This approach could be a problem for extremely low throughput scenarios, but I'm not sure that is a frequent occurrence.